tests/functionaltests/Tests/UIKitTests/UILabelTests.mm

```diff
@@ -0,0 +1,540 @@
+                            // we will re-open these tests until #2159 is closed.
+                            const BOOL skipVerifyTextRectForBounds = YES;
+                            if (!skipVerifyTextRectForBounds) {
+                                ret = CGRectIsWithInDistance(expectedTextRect, actualTextRect, allowedDistance);
+                                EXPECT_TRUE(ret);
+                                if (!ret) {
+                                    textRectForBoundsTestFailureCount++;
+                                }
+                            } else {
+                                textRectForBoundsTestIgnoreCount++;
+                            }
+                        } else {
+                            sizeThatFitTestIgoreCount++;
+                            textRectForBoundsTestIgnoreCount++;
+                        }
+
+                        testId++;
+                    });
+                }
+            }
+        }
+    }
+
+    TEST_METHOD(UILabel_VerifyNumberOfLines) {
+        StrongId<UILabelViewController> labelVC;
+        labelVC.attach([[UILabelViewController alloc] init]);
+        UXTestAPI::ViewControllerPresenter testHelper(labelVC);
+
+        UILabel* label = [labelVC label];
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            // get the backing xaml element, which is Gird contains textBlock
+            WXCGrid* labelGrid = rt_dynamic_cast<WXCGrid>([label xamlElement]);
+            Microsoft::WRL::ComPtr<IInspectable> inspectable(XamlGetLabelTextBlock([labelGrid comObj]));
+            WXCTextBlock* textBlock = _createRtProxy([WXCTextBlock class], inspectable.Get());
+            ASSERT_OBJCNE(textBlock, nil);
+
+            // verify default value is 1
+            ASSERT_TRUE(label.numberOfLines == 1);
+            ASSERT_TRUE(textBlock.maxLines == 1);
+
+            // verify setting to different values other than default
+            label.numberOfLines = 0;
+            ASSERT_TRUE(textBlock.maxLines == 0);
+
+            label.numberOfLines = 3;
+            ASSERT_TRUE(textBlock.maxLines == 3);
+
+            label.numberOfLines = 7;
+            ASSERT_TRUE(textBlock.maxLines == 7);
+
+        });
+    }
+
+    WUColor* ConvertUIColorToWUColor(UIColor* uiColor) {
+        CGFloat r, g, b, a;
+        [uiColor getRed:&r green:&g blue:&b alpha:&a];
+
+        return [WUColorHelper fromArgb:(unsigned char)(a * 255)
+                                     r:(unsigned char)(r * 255)
+                                     g:(unsigned char)(g * 255)
+                                     b:(unsigned char)(b * 255)];
+    }
+
+    UIColor* ConvertWUColorToUIColor(WUColor* wuColor) {
+        CGFloat r = wuColor.r / 255.0;
+        CGFloat g = wuColor.g / 255.0;
+        CGFloat b = wuColor.b / 255.0;
+        CGFloat a = wuColor.a / 255.0;
+        return [UIColor colorWithRed:r green:g blue:b alpha:a];
+    }
```

Comment:
Don't we have these helper functions already in XamlUtilities? We should try and converge them or put them outside of LabelTests at least in case other controls need to use them.